src/app/projects/[projectId]/Requirements.tsx

```diff
@@ -0,0 +1,323 @@
+"use client"; // If this is a client-side component
+
+import { Field, Form, Formik, FieldArray } from "formik";
+import { Button } from "t/components/ui/button";
+import { Label } from "t/components/ui/label";
+import { Textarea } from "t/components/ui/textarea";
+import { Input } from "t/components/ui/input";
+import { api } from "techme/trpc/react"; // Your TRPC client
+import React, { useState } from "react";
+import { useSession } from "next-auth/react";
+
+interface Requirement {
+  id?: number;
+  projectId: number;
+  title: string;
+  description: string;
+  status: string;
+  priority: number;
+  lastModifiedBy: string;
+}
+
+interface RequirementsProps {
+  projectId: number;
+}
+
+const Requirements: React.FC<RequirementsProps> = ({ projectId }) => {
+  const {
+    data: requirements,
+    isLoading,
+    isError,
+  } = api.requirements.getAllRequirements.useQuery({ projectId });
```

Comment:
not to do rn but think about rendering this on server. with all the other components also